cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -614,6 +615,11 @@ public boolean hasType(String datasetTypeName) {
   }
 
   private void publishAudit(Id.DatasetInstance datasetInstance, AuditType auditType) {
+    // Don't publish audit for system datasets admin operations, there can be a deadlock
```

Comment:
tracker should not show system datasets - am I missing something? In tracker, you're always in a namespace, and we don't surface system as a namespace in CDAP.